.circleci/workflows.yml

```diff
@@ -681,6 +685,20 @@ commands:
           steps:
             - ferrocene-setup-darwin
       - aws-oidc-auth
+      - when:
+          condition:
+            equal: [<<parameters.qnx>>, true]
+          steps:
+            - run:
+                name: Fetch QNX710 toolchain
+                command: aws s3 cp s3://ferrocene-ci-artifacts/qnx/qnx710.tar.zst - | zstd -d | tar x
+            - run:
+                name: Set PATH to include QNX710 tools
+                command: echo 'source $(pwd)/qnx710/qnxsdp-env.sh' >> "$BASH_ENV"
+            - run:
+                name: Start QNX License Server port forwarding
+                command: aws ssm start-session --target i-0accef99ca2244097 --document-name AWS-StartPortForwardingSession --parameters '{"portNumber":["56859"], "localPortNumber":["56859"]}'
```

Comment:
I'd add a tag (probably `Name`?) to the instance [and first search the instance by tag](https://serverfault.com/a/560534) to get its ID. That way, if we ever need to recreate the instance for whatever reason we don't need to go and manually change the CI configuration.

Reply:
Given that `Describe*` style actions [can't be limited to specific resources](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-policy-structure.html#ec2-supported-iam-actions-resources) I don't want to do that since it would require permitting the CI token to enumerate all the ec2 nodes of our infrastructure. (It might seem harmless now, but I don't think it's a wise road to travel down)

Perhaps we could have terraform [provision a file to the bucket with the QNX artifacts](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket_object) which gets recreated if the node changes?